ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -27,22 +27,43 @@ import {
 import { Drawer, ErrorAlert } from '@perses-dev/components';
 import { JsonObject } from '@perses-dev/core';
 import { PluginBoundary } from '@perses-dev/plugin-system';
-import { ChangeEvent, FormEvent, useState } from 'react';
+import { ChangeEvent, FormEvent, useEffect, useState } from 'react';
 import { useDashboardApp, useLayouts, usePanels } from '../../context';
 import { removeWhiteSpacesAndSpecialCharacters } from '../../utils/functions';
 import { PanelOptionsEditor, PanelOptionsEditorProps } from './PanelOptionsEditor';
 
+interface PanelDrawerHeaderProps {
+  onClose: () => void;
+  panelRef?: string;
+}
+
 const PanelDrawer = () => {
   const { layouts, addItemToLayout } = useLayouts();
-  const { updatePanel } = usePanels();
+  const { panels, updatePanel } = usePanels();
   const { panelDrawer, closePanelDrawer } = useDashboardApp();
 
-  const [group, setGroup] = useState(0);
-  const [panelName, setPanelName] = useState('');
-  const [panelDescription, setPanelDescription] = useState('');
+  let defaultPanelName = '';
+  let defaultDescription = '';
+  if (panelDrawer?.panelRef) {
+    // editing an existing panel
+    defaultPanelName = panels[panelDrawer.panelRef]?.display.name ?? '';
+    defaultDescription = panels[panelDrawer.panelRef]?.display.description ?? '';
+  }
+  const [group, setGroup] = useState(panelDrawer?.groupIndex);
+  const [panelName, setPanelName] = useState(defaultPanelName);
+  const [panelDescription, setPanelDescription] = useState(defaultDescription);
   const [kind, setKind] = useState('');
   const [options, setOptions] = useState<JsonObject>({});
 
+  useEffect(() => {
+    setGroup(panelDrawer?.groupIndex);
+    if (panelDrawer?.panelRef) {
+      // editing an existing panel
+      setPanelName(panels[panelDrawer.panelRef]?.display.name ?? '');
+      setPanelDescription(panels[panelDrawer.panelRef]?.display.description ?? '');
+    }
+  }, [panelDrawer, panels]);
```

Comment:
Another dumb question: what situation does this effect handle?

Whenever I see an effect like this setting state, I always wonder if it's possible to either move this logic outside of an effect. Sometimes that means to where the state is initialized (i.e. where panel name/description are initialized), or sometimes it means to an event handler where its dependencies are being changed (i.e. where `panelDrawer` and `panels` are changing to trigger it).

Reply:
when editing an existing panel, the panel drawer should display the existing panel's name and description in the text fields

Reply:
yep this component doesn't get remounted when you open/close because otherwise we won't see the cool animation of it sliding in and out